api/src/main/java/org/openmrs/Condition.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import java.util.Date;
+
+/**
+ * This is the condition class.
+ * It records detailed information about a condition, 
+ * problem, diagnosis, or other event, situation or issue.
+ */
+public class Condition extends BaseChangeableOpenmrsData {
+	public static final long serialVersionUID = 1L;
+
+	private Integer conditionId;
+
+	private CodedOrFreeText condition;
+
+	private ConditionClinicalStatus clinicalStatus;
+
+	private ConditionVerificationStatus verificationStatus;
+
+	private Condition previousVersion;
+
+	private String additionalDetail;
+
+	private Date onsetDate;
+
+	/**
+	 * This is a convenience constructor that instantiates a class with the required parameters
+	 *
+	 * @param condition
+	 * @param clinicalStatus
+	 */
+	public Condition(CodedOrFreeText condition, ConditionClinicalStatus clinicalStatus) {
+		this.condition = condition;
+		this.clinicalStatus = clinicalStatus;
+	}
+
+	/**
+	 * Default constructor
+	 */
+	public Condition() {
+	}
+
+	/**
+	 *Getters and setters 
+	 */
+
+	public Integer getConditionId() {
+		return conditionId;
+	}
+
+	public void setConditionId(Integer conditionId) {
+		this.conditionId = conditionId;
+	}
+
+	public CodedOrFreeText getCondition() {
+		return condition;
+	}
+
+	public void setCondition(CodedOrFreeText condition) {
+		this.condition = condition;
+	}
+
+	public ConditionClinicalStatus getClinicalStatus() {
+		return clinicalStatus;
+	}
+
+	public void setClinicalStatus(ConditionClinicalStatus clinicalStatus) {
+		this.clinicalStatus = clinicalStatus;
```

Comment:
Can you do proper javadoc for these methods?

Reply:
Okay. I am on this.